### DISCUSSION TITLE
Can we write a kubernetes operator using kubevela

### DISCUSSION THREAD

**Discussion by gowtham500** [kubevela/kubevela#6191] [Q&A]
Hello team,

Can we write a Kubernetes operator using kubevela ? If so can you please share examples for that.

**Comment by wonderflow** [comment]
What problem you want to solve? You don't need to write K8s operator when using KubeVela.

**Reply by gowtham500** [reply]
My challenge is creating an operator to install DB on k8s. 

Lets say we have a CRD and CR for a database. I give the configuration to create a cluster in the CR - example: replicas : 3 or 5.  When someone creates this CR, usually an operator listens for these events and tries to create them. I have set replicas as 3, the operator creates a 3 node DB cluster. Later if i change the count to 5, the operator spins up another 2 and adds to existing DB cluster.

So basically listen to events and create them.

Can you please let me know how we implement the reconciliation login in kubevela. Can you please share simple examples for that.

**Reply by wonderflow** [reply]
I'm sorry that KubeVela maybe not the right choice in your scenario. You can refer to [kubebuilder](https://kubebuilder.io/) as a operator framework.

There're also projects like kubeblocks solving this kind of scenarios: https://github.com/apecloud/kubeblocks .
